### ISSUE TITLE
linter: false positive for `explicit-module-boundary-types`

### ISSUE THREAD

**Issue by rtarpine-vanta** [oxc-project/oxc#20300]
### What version of Oxlint are you using?

1.55.0

### What command did you run?

oxlint $file.ts

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "rules": {
    "typescript/explicit-module-boundary-types": "error"
  },
}
```

### What happened?

The following code triggers two `typescript-eslint(explicit-module-boundary-types)` errors for `f`: missing argument type and missing return type.
```ts
type F = (x: number) => number;

export const f = (x => x) satisfies F;
```
But the original `@typescript-eslint/explicit-module-boundary-types` rule doesn't complain since `f` is wrapped with `satisfies`.
